src/expression-to-formula.js

```diff
@@ -26,19 +26,21 @@ export default function expressionToFormula(expression) {
 
   if (operator === 'literal') {
     const arg = expression[1];
-    if (!Array.isArray(arg)) {
-      throw new Error(
-        'Only array arguments are supported for the literal expression'
-      );
+    if (Array.isArray(arg)) {
+      return `${operator}(${stringifyLiteralArray(arg)})`;
     }
-    return `${operator}(${stringifyLiteralArray(arg)})`;
   }
 
   const args = expression.slice(1).map(arg => {
     if (typeof arg === 'string') {
       return `"${arg}"`;
     }
-    if (!Array.isArray(arg)) {
+
+    const isArray = Array.isArray(arg);
+    if (typeof arg === 'object' && !!arg && !isArray) {
+      return JSON.stringify(arg);
```

Comment:
🤔  maybe we should do some pretty-printing here? Maybe add spaces?

Reply:
spaces!